devices/examples/YarpCameraBridgeArucoTestDevice/src/YarpCameraBridgeArucoTestDevice.cpp

```diff
@@ -0,0 +1,185 @@
+
+    // the logging message are streamed using yarp
+    BipedalLocomotion::TextLogging::LoggerBuilder::setFactory(
+        std::make_shared<BipedalLocomotion::TextLogging::YarpLoggerFactory>());
+}
+
+YarpCameraBridgeArucoTestDevice::~YarpCameraBridgeArucoTestDevice() = default;
+
+bool YarpCameraBridgeArucoTestDevice::open(yarp::os::Searchable& config)
+{
+    auto params = std::make_shared<ParametersHandler::YarpImplementation>(config);
+
+    double devicePeriod{0.01};
+    if (params->getParameter("sampling_period_in_s", devicePeriod))
+    {
+        this->setPeriod(devicePeriod);
+    }
+
+    if (!this->setupRobotCameraBridge(params->getGroup("RobotCameraBridge")))
+    {
+        return false;
+    }
+
+    if (!this->setupArucoDetector(params->getGroup("ArucoDetector")))
+    {
+        return false;
+    }
+
+    return true;
+}
+
+
+bool YarpCameraBridgeArucoTestDevice::setupRobotCameraBridge(
+    std::weak_ptr<const ParametersHandler::IParametersHandler> params)
+{
+    constexpr auto logPrefix = "[YarpCameraBridgeArucoTestDevice::setupRobotSensorBridge]";
+
+    auto ptr = params.lock();
+    if (ptr == nullptr)
+    {
+        log()->error("{} The parameters handler is not valid.", logPrefix);
+        return false;
+    }
+
+    m_robotCameraBridge = std::make_unique<YarpCameraBridge>();
+    if (!m_robotCameraBridge->initialize(ptr))
+    {
+        log()->error("{} Unable to configure the 'YarpCameraBridge'", logPrefix);
+        return false;
+    }
+
+
+    return true;
+}
+
+bool YarpCameraBridgeArucoTestDevice::setupArucoDetector(
+    std::weak_ptr<const ParametersHandler::IParametersHandler> params)
+{
+    constexpr auto logPrefix = "[YarpCameraBridgeArucoTestDevice::setupArucoDetector]";
+
+    auto ptr = params.lock();
+    if (ptr == nullptr)
+    {
+        log()->error("{} The parameters handler is not valid.", logPrefix);
+        return false;
+    }
+
+    if (!m_arucoDetector.initialize(ptr))
+    {
+        log()->error("{} Unable to configure the 'ArucoDetector'", logPrefix);
+        return false;
+    }
+
+    return true;
+}
+
+bool YarpCameraBridgeArucoTestDevice::attachAll(const yarp::dev::PolyDriverList& poly)
+{
+    constexpr auto logPrefix = "[YarpCameraBridgeArucoTestDevice::attachAll]";
+    bool ok = true;
+    if (!m_robotCameraBridge->setDriversList(poly))
+    {
+        log()->error("{} Could not attach drivers list to sensor bridge.", logPrefix);
+        return false;
+    }
+
+    if (ok)
+    {
+        return start();
+    }
+    return ok;
+}
+
+void YarpCameraBridgeArucoTestDevice::run()
+{
+    constexpr auto logPrefix = "[YarpCameraBridgeArucoTestDevice::run]";
+
+    const double time = BipedalLocomotion::clock().now().count();
+    double imgRcvTime;
+    bool ok = m_robotCameraBridge->getColorImage("rgbd_cam", m_rgbImg, imgRcvTime);
+    if (ok)
+    {
+        cv::imshow("rgb", m_rgbImg);
+        cv::waitKey(1);
```

Comment:
This is just a curiosity, what `cv::waitKey()` does?  and why this is required?

Reply:
waitKey simply adds a delay in milliseconds and a small delay is necessary to be added for `imshow` to display the image in the corresponding window. For more information, see https://docs.opencv.org/2.4/modules/highgui/doc/user_interface.html#imshow.